datastream-server/src/test/java/com/linkedin/datastream/server/assignment/TestLoadBasedPartitionAssignmentStrategy.java

```diff
@@ -100,6 +106,45 @@ public void fallbackToBaseClassWhenElasticTaskAssignmentDisabledTest() {
     Mockito.verify(mockProvider, times(0)).getThroughputInfo(any(String.class));
   }
 
+  @Test
+  public void fallbackToBaseClassWhenThroughputFetchFailsTest() {
+    PartitionThroughputProvider mockProvider = mock(PartitionThroughputProvider.class);
+    Mockito.when(mockProvider.getThroughputInfo(any(DatastreamGroup.class))).thenThrow(new RuntimeException());
+    boolean enableElasticTaskAssignment = true;
+    Optional<Integer> maxTasks = Optional.of(100);
+    Optional<Integer> imbalanceThreshold = Optional.of(50);
+    Optional<Integer> maxPartitionPerTask = Optional.of(100);
+    Optional<Integer> partitionsPerTask = Optional.of(50);
+    Optional<Integer> partitionFullnessFactorPct = Optional.of(80);
+    Optional<Integer> taskCapacityMBps = Optional.of(5);
+    Optional<Integer> taskCapacityUtilizationPct = Optional.of(90);
+    Optional<Integer> throughputInfoFetchTimeoutMs = Optional.of(1000);
+    Optional<Integer> throughputInfoFetchRetryPeriodMs = Optional.of(200);
+    Optional<ZkClient> zkClient = Optional.of(_zkClient);
+
+    LoadBasedPartitionAssignmentStrategy strategy = Mockito.spy(new LoadBasedPartitionAssignmentStrategy(mockProvider,
+        maxTasks, imbalanceThreshold, maxPartitionPerTask, enableElasticTaskAssignment, partitionsPerTask,
+        partitionFullnessFactorPct, taskCapacityMBps, taskCapacityUtilizationPct, throughputInfoFetchTimeoutMs,
+        throughputInfoFetchRetryPeriodMs, zkClient, _clusterName));
```

Comment:
nit: You can extract the strategy creation option to a separate method to avoid duplicate code. You can use Intellij utility to do the refactoring for you.

Reply:
I'd prefer to leave it as it is. If I move the strategy creation in a separate method, I'll have to pass a lot of arguments in each call and it's prone to errors.